task_02/shamil_latypov/graph.cpp

```diff
@@ -0,0 +1,127 @@
+#include <iostream>
+#include <stdlib.h>
+#include <map>
+#include <string>
+#include <vector>
+#include <algorithm>
+#include <fstream>
+#include <sstream>
+
+struct edge {
+    int id = 0;
+    //std::string name = "";
+};
+
+struct vertex {
+    int id = 0;
+    
+    friend bool operator< (vertex a, vertex b) {
+        return a.id < b.id;
+    }
+    
+    friend bool operator== (vertex a, vertex b) {
+        return a.id == b.id;
+    }
+
+    //std::string name = "";
+};
+
+struct graph {
+    struct vertex v1;
+    struct edge ed;
+    struct vertex v2;
+
+    friend bool operator< (graph a, graph b) {
+        if (a.v1 == b.v1) return a.v2 < b.v2;
+        else return a.v1 < b.v1;
+    }
+};
```

Comment:
Ты наверное, пытался описать здесь структуру `Edge`, а не `Graph`, потому что `Graph` в данном виде не имеет смысла, он должен содержать в себе много вершин и много граней.
Что-то типа:
```suggestion
struct Edge {
    EdgeId id;
    VertexId v1_id;
    VertexId v2_id;
};
```